mobile/src/main/java/org/openhab/habdroid/ui/WidgetAdapter.kt

```diff
@@ -610,6 +621,16 @@ class WidgetAdapter(
             labelView.text = widget.label + label
             labelView.applyWidgetColor(widget.valueColor, colorMapper)
             labelView.isGone = widget.label.isEmpty()
+            val isNested = parentWidget?.type == Widget.Type.Frame
+
+            val paddingTop = if (isNested) 0 else originalPaddingTop
+            containerView.updatePadding(top = paddingTop, bottom = paddingTop)
+
+            backgroundColorMap[isNested]?.let { (containerView.background as GradientDrawable).setColor(it) }
+
+            labelView.layoutParams = (labelView.layoutParams as ViewGroup.MarginLayoutParams).apply {
+                marginStart = if (isNested) labelView.resources.dpToPixel(8f).toInt() else 0
```

Comment:
I don't know about @mueller-ma's stance on this, but personally I don't like manually messing with layout properties like this. I think having separate layouts for both type of frames, a common base class bundling the functionality and separate derived classes only differing in the constructor parameters (layout IDs) would be a bit more effort, but also be more maintainable in the long run.

Reply:
I agree

Reply:
I can have a look into the suggested implementation.

Reply:
@jimtng If you want to consider picking: https://github.com/maniac103/openhab.android/commit/b51e2a1899337c9c904bb5ad3482c69748761f6f

I also shrinked font size of secondary frames there.

Normal:
![normal](https://github.com/user-attachments/assets/9c563488-da2d-4311-8149-994520bdbf70)

Compact:
![compact](https://github.com/user-attachments/assets/a869c347-76b4-4188-98d3-478c09ebd650)

Reply:
I just noticed I forgot the text indention in the compact layout. Fixed in https://github.com/maniac103/openhab.android/commit/59da677e95ee92521c3a0bcb203e13a10cd127de

Reply:
@maniac103 thanks!